src/shared/utilities/childProcess.ts

```diff
@@ -95,8 +94,15 @@ export class ChildProcess {
         this.log = options.logging !== 'no' ? logger.getLogger() : logger.getNullLogger()
     }
 
-    // TODO: add a way to create 'base' child process instances without necessarily needing to construct a new one each time
-    // this way modules can define default parameters (e.g. `Timeout`) and have all downstream functions use it
+    // Inspired by 'got'
+    public static extend(options: ChildProcessOptions) {
+        return class extends this {
+            public constructor(command: string, args: string[] = []) {
+                super(command, args, options)
+            }
+        }
+    }
```

Comment:
What value does this provide? Should we be dealing with generic classes when you can just define an extending class manually?

Reply:
Oh I must've missed this feedback. 

The goal here is to have narrowly scoped, specialized forms of `ChildProcess`. Classical inheritance is great, except for the fact that it's easy to abuse. `extend` allows me to create a class that captures the options but is also an instance of `ChildProcess`. But I haven't polluted the namespace with a bunch of sub-classes since the new classes are anonymous.